californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java

```diff
@@ -196,10 +202,17 @@ public void receiveRequest(final Exchange exchange, final Request request) {
 
 		if (request.isDuplicate()) {
 			// Request is a duplicate, so resend ACK, RST or response
-			if (exchange.getCurrentResponse() != null) {
+			Response currentResponse = exchange.getCurrentResponse();
+			if (currentResponse != null) {
 				LOGGER.debug("respond with the current response to the duplicate request");
+				if (currentResponse.isConfirmable()) {
+					// Simulate retransmission cycle, CON responses requires this
+					exchange.setFailedTransmissionCount(1);
```

Comment:
I'm not sure to understand this line ? Why setting it to `1` ?

Reply:
````
java.lang.IllegalArgumentException: message with already registered ID [41845] is not a re-transmission, cannot register exchange
	at org.eclipse.californium.core.network.InMemoryMessageExchangeStore.registerWithMessageId(InMemoryMessageExchangeStore.java:212)
	at org.eclipse.californium.core.network.InMemoryMessageExchangeStore.registerOutboundResponse(InMemoryMessageExchangeStore.java:323)
	at org.eclipse.californium.core.network.UdpMatcher.sendResponse(UdpMatcher.java:144)
	at org.eclipse.californium.core.network.CoapEndpoint$OutboxImpl.sendResponse(CoapEndpoint.java:694)
	at org.eclipse.californium.core.network.stack.BaseCoapStack$StackBottomAdapter.sendResponse(BaseCoapStack.java:197)
	at org.eclipse.californium.core.network.stack.ReliabilityLayer.receiveRequest(ReliabilityLayer.java:215)
	at org.eclipse.californium.core.network.stack.AbstractLayer.receiveRequest(AbstractLayer.java:81)
	at org.eclipse.californium.core.network.stack.BaseCoapStack.receiveRequest(BaseCoapStack.java:99)
````

Running tests for the leak issue #429, I got this exception and the initial intention of this PR was to fix exactly that exception. I documented that therefore in the header

>    Achim Kraus (Bosch Software Innovations GmbH) - set retransmission for
>                                                    CON responses sent again for
>                                                    duplicated requests.
>                                                    Changed to pass check in 
>                                                    InMemoryMessageExchangeStore
>                                                    registerWithMessageId